### PR TITLE
[PM-1826] [PM-2168] [Tech debt] Migrate file-password-prompt to dialog

### PR DESCRIPTION
**Pull request by djsmith85**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective
Migrate file-password-prompt to dialog which also cleaned up any custom tailwind class usage

## Code changes
- **apps/web/src/app/tools/import-export/dialog/file-password-prompt.component.html:** Replace to use bitDialog
- **apps/web/src/app/tools/import-export/dialog/file-password-prompt.component.html:** Replace usage of ModalRef to DialogRef
- **apps/web/src/app/tools/import-export/import.component.ts:** Get result from new dialog
- **libs/importer/src/importers/bitwarden/bitwarden-password-protected-importer.ts:** Fix issue with cancel/empty password returned

## Screenshots
**BEFORE:**
image

**AFTER:**
image

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Review by aj-rosado — DISMISSED** [review]
Changes are looking good to me!

**Review by aj-rosado — DISMISSED** [review]

File: apps/web/src/app/tools/import-export/dialog/file-password-prompt.component.html [hunk 1]
```diff
@@ -1,45 +1,30 @@
...
   </div>
-</div>
+
+  <ng-container bitDialogFooter>
+    <button bitButton buttonType="primary" type="submit">
+      <span>{{ "importData" | i18n }}</span>
+    </button>
+    <button bitButton bitDialogClose buttonType="secondary" type="button" (click)="cancel()">
+      <span>{{ "cancel" | i18n }}</span>
+    </button>
```
Comment (Hinton) [inline apps/web/src/app/tools/import-export/dialog/file-password-prompt.component.html:28]: I don't think you need the click handler here. `bitDialogClose` should handle it for you.

Comment (djsmith85) [reply apps/web/src/app/tools/import-export/dialog/file-password-prompt.component.html:28]: Nice catch!!

**Review by danielleflinn — CHANGES_REQUESTED** [review]
@djsmith85 looks like we lost the spacing between the instructional text and the text field. Can we add that back in? `tw-m-6` should work well.

**Review by danielleflinn — CHANGES_REQUESTED** [review]
@djsmith85 margin should only be above the field.

**Review by danielleflinn — APPROVED** [review]

**Review by Hinton — APPROVED** [review]

**Review by aj-rosado — APPROVED** [review]

File: apps/web/src/app/tools/import-export/dialog/file-password-prompt.component.html [hunk 2]
```diff
@@ -1,45 +1,30 @@
...
+      />
+      <button type="button" bitSuffix bitIconButton bitPasswordInputToggle></button>
+    </bit-form-field>
   </div>
-</div>
+
+  <ng-container bitDialogFooter>
+    <button bitButton buttonType="primary" type="submit" (click)="submit()">
+      <span>{{ "importData" | i18n }}</span>
+    </button>
```
Comment (Hinton) [inline apps/web/src/app/tools/import-export/dialog/file-password-prompt.component.html:25]: Should probably wrap the input and submit button in a form instead, and use the form submit.

**Review by Hinton — APPROVED** [review]

**Review by aj-rosado — APPROVED** [review]

**Pull request merged**